include/convex_bodies/ellipsoid.h

```diff
@@ -0,0 +1,288 @@
+// VolEsti (volume computation and sampling library)
+
+// Copyright (c) 2012-2018 Vissarion Fisikopoulos
+// Copyright (c) 2018 Apostolos Chalkis
+// Copyright (c) 2021 Vaibhav Thakkar
+
+//Contributed and/or modified by Apostolos Chalkis, as part of Google Summer of Code 2018 program.
+//Contributed and/or modified by Repouskos Panagiotis, as part of Google Summer of Code 2019 program.
+//Contributed and/or modified by Vaibhav Thakkar, as part of Google Summer of Code 2021 program.
+
+// Licensed under GNU LGPL.3, see LICENCE file
+
+
+#ifndef ELLIPSOIDS_H
+#define ELLIPSOIDS_H
+
+#include <iostream>
+#include <Eigen/Eigen>
+#include <boost/math/special_functions/gamma.hpp>
+
+
+template <typename NT>
+NT log_gamma_function(NT x)
+{
+    if (x <= NT(100)) return std::log(tgamma(x));
+    return (std::log(x - NT(1)) + log_gamma_function(x - NT(1)));
+}
+
+
+template <class Point, class MT>
+class Ellipsoid{
+private:
+    typedef typename Point::FT NT;
+    typedef typename std::vector<NT>::iterator viterator;
+    typedef Eigen::Matrix<NT, Eigen::Dynamic, 1> VT;
+
+    // representation is (x - c)' A (x - c) <= 1, center is assumed to be origin for now
+    MT A;
```

Comment:
What about member variables starting with "_" ?

Reply:
changed non-core members variables (other than A, c) to have "_" before their names.